specs/PushNotifications/ToastNotifications-spec.md

```diff
@@ -0,0 +1,551 @@
+    };
+
+    runtimeclass AppNotification
+    {
+        AppNotification(Windows.Data.Xml.Dom.XmlDocument payload);
+
+        // Unique identifier used to replace a notification within a group.
+        String Tag;
+
+        // Unique identifier for a Notification group in the app
+        String Group;
+
+        // A unique identifier for the Notification generated by the platform.
+        UInt32 Id { get; };
+
+        // The notification Xml Payload
+        Windows.Data.Xml.Dom.XmlDocument Payload{ get; };
+
+        // Gets or sets additional information about the Notification progress.
+        AppNotificationProgressData Progress;
+
+        // Gets or sets the time after which a Notification should not be displayed.
+        Windows.Foundation.DateTime Expiration;
+
+        // Indicates whether the Notification will remain in the Action Center after a reboot.
+        Boolean ExpiresOnReboot;
+
+        // Gets or sets the priority for a Notification.
+        // Hints on how and at what urgency level a notification should be presented to the user (whether to wake up the screen, etc).
+        AppNotificationPriority Priority;
+
+        // Gets or sets whether a Notification's pop-up UI is displayed on the user's screen.
+        Boolean SuppressDisplay;
+    }
+
+    // The manager class which encompasses all App Notification API Functionality
+    [experimental]
+    runtimeclass AppNotificationManager
+    {
+        // Gets a Default instance of a AppNotificationManager
+        static AppNotificationManager Default{ get; };
+
+        // Register an activator using an ActivationInfo context and caches the token for unregister
+        void RegisterActivator(AppNotificationActivationInfo details);
+
+        // Unregisters the activator and removes the cached Registration token.
+        void UnregisterActivator();
+
+        // Event handler for Notification Activations
+        event Windows.Foundation.TypedEventHandler<Microsoft.Windows.AppLifecycle.ExtendedActivationKind, AppNotificationActivatedEventArgs> AppNotificationActivated;
+
+        // Displays the Notification in Action Center
+        void Show(AppNotification notification);
+
+        // Updates the Notification for a Progress related operation using Tag and Group
+        Windows.Foundation.IAsyncOperation<AppNotificationProgressResult> UpdateProgressDataAsync(AppNotificationProgressData data, String tag, String group);
+
+        // Updates the Notification for a Progress related operation using Tag
+        Windows.Foundation.IAsyncOperation<AppNotificationProgressResult> UpdateProgressDataAsync(AppNotificationProgressData data, String tag);
+
+        // Get the Notification Setting status for the app
+        AppNotificationSetting Enablement { get; };
```

Comment:
>Enablement [](http://example.com/codeflow?start=31&length=10)

AccessStatus is what we traditionally call this.

Reply:
DISCUSSION: ToastNotifier calls it Setting.

CONSIDER: Setting

Reply:
Incorporated feedback